### ISSUE TITLE
Deprecate assert_str_equal

### ISSUE THREAD

**Issue by dstansby** [matplotlib/matplotlib#8112]
Judging by codecov, `assert_str_equal` in `tests/__init__.py` is never used (https://codecov.io/gh/matplotlib/matplotlib/src/master/lib/matplotlib/tests/__init__.py#L22), and grep agrees:

```
$ grep -r 'assert_str_equal' lib/
lib//matplotlib/tests/__init__.py:def assert_str_equal(reference_str, test_str,
```

So I don't see why it shouldn't be deprecated and then removed.

**Comment by tacaswell** [comment]
I think there may be a PR in flight to do that, possibly from @dopplershift ?

**Comment by anntzer** [comment]
I doubt so given that `assert_str_equal` was still in use until *very* recently (removed by @QuLogic in https://github.com/matplotlib/matplotlib/pull/8026).

**Comment by QuLogic** [comment]
Fixed by #8118.
